js/sidenav.js

```diff
@@ -55,12 +55,113 @@ $('.sidenav').on('click', 'a', function (e) {
   });
 });
 
-//var navHeight = function () {
-//  $('.sidenav').css('height', (window.innerHeight - 50));
-//};
-//
-//navHeight();
-//
-//$( window ).resize(function () {
-//  navHeight();
-//});
+// In-page navigation
+
+function ready(fn) {
+  document.addEventListener('DOMContentLoaded', fn, false);
+}
+
+ready(() => {
+  const motionQuery = window.matchMedia('(prefers-reduced-motion)');
+
+  const TableOfContents = {
+    container: document.querySelector('.in-page-navigation'),
+    links: null,
+    headings: null,
+    intersectionOptions: {
+      rootMargin: '0px',
+      threshold: 1,
+    },
+
+    previousSection: null,
+    observer: null,
+
+    init() {
+      this.handleObserver = this.handleObserver.bind(this);
+
+      this.setUpObserver();
+      this.findLinksAndHeadings();
+      this.observeSections();
+
+      this.links.forEach((link) => {
+        link.addEventListener('click', this.handleLinkClick.bind(this));
+      });
+    },
+
+    handleLinkClick(evt) {
+      evt.preventDefault();
+      let id = evt.target.getAttribute('href').replace('#', '');
```

Comment:
Optional, you can should be able to get the href `evt.target.hash`, instead of `getAttribute('href')`.

See MDN reference - https://developer.mozilla.org/en-US/docs/Web/API/Location/hash